### PR TITLE
[sdk-metrics] Add support for .NET 9 Advice API

### LINKED ISSUE #5487: Support .NET 9 Advice API
This is a tracking issue for supporting https://github.com/dotnet/runtime/issues/63650 once it is available in .NET 9.

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Fixes #5487
Design discussion issue #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [X] Unit tests added/updated
* [X] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Comment by cijothomas** [comment]
@rajkumar-rangaraj Do you know if httpclient, aspnetcore started leveraging the advice for providing custom buckets ?

**Comment by rajkumar-rangaraj** [comment]
> @rajkumar-rangaraj Do you know if httpclient, aspnetcore started leveraging the advice for providing custom buckets ?

I haven't checked this part yet.

File: test/OpenTelemetry.Tests/Metrics/MetricViewTests.cs [hunk 1]
```diff
@@ -570,6 +570,130 @@ public void ViewToProduceCustomHistogramBound()
...
+            Assert.Equal(expectedBucketCounts[index], histogramMeasurement.BucketCount);
+            index++;
+            actualCount++;
+        }
+
+        Assert.Equal(boundaries.Count + 1, actualCount);
+    }
+
+    [Fact]
+    public void ViewTakesPrecedenceOverAdviceToProduceCustomHistogramBound()
```
Comment (CodeBlanch) [inline test/OpenTelemetry.Tests/Metrics/MetricViewTests.cs:635]: Lot of duplication between this test and the one above. Should we convert it into a theory?

```csharp
    [Theory]
    [InlineData(false)]
    [InlineData(true)]
    public void HistogramWithAdviceBoundariesSpecifiedTests(bool useViewToOverride) {}
```

**Comment by CodeBlanch** [comment]
Definitely need a CHANGELOG for this one 😄

**Comment by cijothomas** [comment]
> > @rajkumar-rangaraj Do you know if httpclient, aspnetcore started leveraging the advice for providing custom buckets ?
> 
> I haven't checked this part yet.

np! We just need to make sure we don't remove the hardcoded buckets that exist today in sdk, until that happens and old versions are out of support.

**Comment by cijothomas** [comment]
> Definitely need a CHANGELOG for this one 😄

I also suggest to document this near https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/metrics/customizing-the-sdk#explicit-bucket-histogram-aggregation and recommend users to leverage Advice, wherever feasible

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry/CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,15 @@ Notes](../../RELEASENOTES.md).
   `Microsoft.Extensions.Diagnostics.Abstractions` packages version to
   `9.0.0-rc.1.24431.7`.
   ([#5853](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5853))
+  
+* Added support in metrics for histogram bucket boundaries set via the .NET 9
+  [InstrumentAdvice<T>](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.instrumentadvice-1)
```
Comment (CodeBlanch) [inline src/OpenTelemetry/CHANGELOG.md:21]:
```suggestion
* Added support in metrics for histogram bucket boundaries set via the .NET 9
  [InstrumentAdvice&lt;T&gt;](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.instrumentadvice-1)
```

File: src/OpenTelemetry/CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +16,15 @@ Notes](../../RELEASENOTES.md).
   `Microsoft.Extensions.Diagnostics.Abstractions` packages version to
   `9.0.0-rc.1.24431.7`.
   ([#5853](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5853))
+
```
Comment (reyang) [inline src/OpenTelemetry/CHANGELOG.md:19]:
```suggestion

```

**Pull request merged**